merkledag/merkledag.go

```diff
@@ -153,7 +153,10 @@ type sesGetter struct {
 
 func (sg *sesGetter) Get(ctx context.Context, c *cid.Cid) (node.Node, error) {
 	blk, err := sg.bs.GetBlock(ctx, c)
-	if err != nil {
+	switch err {
+	case bserv.ErrNotFound:
+		return nil, ErrNotFound
+	default:
```

Comment:
You're missing the case where `err` is nil (this is what's causing the segfaults in CI).